jobqueue_features/decorators.py

```diff
@@ -173,6 +173,11 @@ def _get_cluster_attribute(self, cluster, attribute, default, **kwargs):
     def _submit(self, cluster, client, f, *args, **kwargs):
         # For MPI tasks, let's assume functions are not pure (by default)
         pure, kwargs = self._get_cluster_attribute(cluster, "pure", False, **kwargs)
+        # Handle the case where the cluster had a pure attribute
+        if hasattr(cluster, "pure") and pure is None:
```

Comment:
Maybe I should be checking the value of `cluster.pure` rather than assuming it was `None`

Reply:
So are we updating the cluster attributes from the task? Should we set the `pure` attribute back to previous at the end of the task? 

Also - we could go for `cluster.pure = bool(getattr(cluster, "pure", False))`

Reply:
Hmm, you are right, maybe submit is not the right place to set this value at all. We should be doing it when the cluster is declared.

Reply:
and the value set at that time should persist but can be overridden by a kwarg.